[chore][receiver/sqlquery] Move reusable logic to internal package

crobert-1 opened the pull request:
**Description:** <Describe what has changed.>
<!--Ex. Fixing a bug - Describe the bug and how this fixes the issue.
Ex. Adding a feature - Explain what this achieves.-->
As noted in the related issues, a lot of the logic in the sqlquery receiver can be moved to a central package to be used by other receivers. I realize this appears to be a large change, but it's purely a re-organization PR, there's no functional changes. GitHub is not recognizing moved files, so please refer to new file names and their deleted counterparts. A lot of members are now public in the internal package that were private in the receiver which is likely what's causing GitHub to miss that they're just moved files.

This doesn't require a changelog (in my mind) because all things moving were originally private to the receiver. They're now still purely internal, so they won't be published.

I believe `internal` is the best destination for now as it allows us to solidify the usage and interface before moving it to `pkg` to be publicly available. There are still a lot of unknowns as far as how it can be used by other receivers, so I fully expect "breaking changes" to this. If we keep it internal we can update all usages with the changing interface, so the changes won't be breaking.

**Link to tracking Issue:** <Issue number if applicable>
#30297, #13546

**Testing:** <Describe what testing was performed and which tests were added.>
Existing tests are all passing.

**Documentation:** <Describe the documentation added.>
This is purely an internal change.

crobert-1 commented on receiver/sqlqueryreceiver/config.go:
```diff
@@ -4,186 +4,29 @@
-		return nil
-	}
-	return fmt.Errorf("metric config has unsupported aggregation: '%s'", a)
+	return c.Config.Validate()
```
I'd appreciate feedback on naming that would make sense here. `c.Config` is pretty clunky, but I'm not sure how best to name it since it's a config embedded within a config.

dmitryax replied:
Calling `Validate()` is not needed. It's called automatically on all the embedded structs

crobert-1 replied:
Good to know, I've removed and tests are still passing.

dmitryax commented on .github/CODEOWNERS:
```diff
@@ -124,6 +124,7 @@ internal/kubelet/                                        @open-telemetry/collect
 internal/metadataproviders/                              @open-telemetry/collector-contrib-approvers @Aneurysm9 @dashpole
 internal/sharedcomponent/                                @open-telemetry/collector-contrib-approvers @open-telemetry/collector-approvers
 internal/splunk/                                         @open-telemetry/collector-contrib-approvers @dmitryax
+internal/sqlquery/                                       @open-telemetry/collector-contrib-approvers @dmitryax @pmcollins
```
Can you add yourself as a code owner?
```suggestion
internal/sqlquery/                                       @open-telemetry/collector-contrib-approvers @crobert-1 @dmitryax
```

crobert-1 replied:
Done

crobert-1 committed:
[chore][receiver/sqlquery] Move re-usable logic to central package

crobert-1 committed:
make gotidy

crobert-1 committed:
Changes requested in PR

- Add myself as code owner to new package
- Remove unrelated godep updates
- Remove unnecessary Validate() call

crobert-1 commented on cmd/configschema/go.mod:
```diff
@@ -675,14 +673,14 @@ require (
 	golang.org/x/term v0.16.0 // indirect
 	golang.org/x/time v0.5.0 // indirect
-	golang.org/x/tools v0.16.0 // indirect
+	golang.org/x/tools v0.17.0 // indirect
```
I think module versions were updated when creating the new `go.mod` for the new `internal/sqlquery` package, and all of the versions got updated throughout the project as a result. I've removed unrelated updates. 👍

dmitryax commented:
@crobert-1 can you please rebase and check CI failures?

crobert-1 committed:
make gotidy

crobert-1 committed:
Update deps

crobert-1 committed:
make gotidy

crobert-1 committed:
Merge branch 'main' into sqlquery_package

dmitryax commented on receiver/sqlqueryreceiver/config.go:
```diff
@@ -4,186 +4,29 @@
+	sqlquery.Config `mapstructure:",squash"`
 }
 
 func (c Config) Validate() error {
```
This function is redundant now. feel free to remove in a follow-up PR
